src/connection_string.ts

```diff
@@ -309,10 +300,58 @@ export function parseOptions(
     Object.entries(options).filter(([, v]) => v != null)
   );
 
+  // Validate options that can only be provided by one of uri or object
+
+  if (urlOptions.has('serverApi')) {
+    throw new MongoParseError(
+      'URI cannot contain `serverApi`, it can only be passed to the client'
+    );
+  }
+
   if (objectOptions.has('loadBalanced')) {
     throw new MongoParseError('loadBalanced is only a valid option in the URI');
   }
 
+  // SRV connection string validations
+
+  const nonZeroSrvMaxHosts =
+    objectOptions.get('srvMaxHosts') > 0 || urlOptions.get('srvMaxHosts') > 0;
```

Comment:
I think this is better to do after the options are parsed because this is really only for checking against the replicaSet (see my other comment on the redundant loadbalanced check), if we do this just before the loadBalanced validation below, we'll be able to read the srvMaxHosts from the already parsed options (instead of trying to manually read them and bypassing the validation on the option definitions themselves; this would also ensure the correct precedence is maintained if option is specified in more than one place: in the above logic, if srvMaxHosts is set to 0 in the object but not in the uri, we would throw an error, even tough the objectOptions should override it and not throw).